### PR TITLE
Resolve Hive API deprecation warnings

### PR DESCRIPTION
**Pull request by gerlowskija**
Prior to this commit, several classes used deprecated Hive APIs, that
ended up being removed in the Hive 2.x release line.  This commit fixes
a few of these warnings, to clear the way for use of Hive 2.x releases
coming soon.  Specifically, the changes involved:

- changing references to "Constants" to its successor "serdeConstants"
- using the "AbstractSerDe" class as a supertype, instead of the "SerDe"
  interface

### CODE REVIEW

**Comment by ctargett** [comment]
With this PR, would it be impossible for the SerDe to work with Hive 1.x? Just curious.

**Comment by gerlowskija** [comment]
Nope.  This PR still keeps the door open on Hive 1.x.  All it does is replace some client APIs we were calling that had been deprecated in the 1.x line.

So it makes it a bit easier to upgrade our Hive version at some future point.  But there's no hive upgrade in this PR itself.

File: solr-hive-core/src/main/java/com/lucidworks/hadoop/hive/LWSerDe.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 // deprecation -> SerDe
 @SuppressWarnings("deprecation")
```
Comment (acesar) [inline solr-hive-core/src/main/java/com/lucidworks/hadoop/hive/LWSerDe.java]: `SuppressWarnings("deprecation")` should we remove this one now?

File: solr-hive-core/src/main/java/com/lucidworks/hadoop/hive/LWStorageHandler.java [hunk 2]
```diff
@@ -34,7 +35,7 @@
 
   @Override
   @SuppressWarnings("deprecation")
```
Comment (acesar) [inline solr-hive-core/src/main/java/com/lucidworks/hadoop/hive/LWStorageHandler.java]: SuppressWarnings("deprecation") should we remove this one now?

**Pull request merged**